tiledb/CMakeLists.txt

```diff
@@ -592,8 +592,24 @@ endif()
 
 if (TILEDB_GCS)
   find_package(google_cloud_cpp_storage CONFIG REQUIRED)
-  target_link_libraries(TILEDB_CORE_OBJECTS_ILIB INTERFACE CURL::libcurl google-cloud-cpp::storage google-cloud-cpp::experimental-storage-grpc)
-  install_all_target_libs(google-cloud-cpp::storage)
+  target_link_libraries(TILEDB_CORE_OBJECTS_ILIB INTERFACE
```

Comment:
Is there still no target for this? -- any way to avoid listing out all the libraries individually?

Reply:
I'm not sure what you mean by "this". And (at least AFAIK) explicitly listing our transitive dependencies is necessary for the bundled `TileDB::tiledb_static` target in GitHub Releases to work.

Even if we could figure out a way to automatically walk the dependency graph of the `google-cloud-cpp:storage` target and bundle all referenced libraries[^1], we risk bundling static libraries we don't use, like the 85 `absl_***.lib` files, or the 800MB `grpc.lib` that (fortunately) is not actually used.

[^1]: I've considered it in the past but could not think of a reliable way to do this; as stated in other places, we are fighting CMake by bundling the static libraries of our dependencies.

Reply:
> I'm not sure what you mean by "this"

Sorry, I meant something like [`AWSSDK_LINK_LIBRARIES`](https://github.com/TileDB-Inc/TileDB/blob/dd0006a13791016e898071ca93734ef2d4524032/tiledb/CMakeLists.txt#L956-L958). If not, no problem.

Reply:
From a quick look there doesn't seem to be such thing; [the CMake config file for the GCS SDK](https://github.com/googleapis/google-cloud-cpp/blob/37f726cde8bf3f246b3cddb3d5385d7ccc08a07a/google/cloud/storage/config.cmake.in) is pretty simple compared to what the AWS SDK does.